### ISSUE TITLE
image process for training should not div 255？, I didn't find this operation in your code.

### ISSUE THREAD

**Issue by tantao258** [YunYang1994/tensorflow-yolov3#16]

**Comment by YunYang1994** [comment]
Sorry. I'll fix it

**Comment by tantao258** [comment]
when you compute loss_class, I found that the logits have operated by sigmoid function , maybe there is a error

**Comment by YunYang1994** [comment]
@hojjatabdollahi you mean you have trained it with COCO dataset? what results do you have now?

**Comment by tantao258** [comment]
@YunYang1994 yester day I began to train model on coco dataset, In 18 hours, the loss went from 200 to 20, the training is gonging on.

**Comment by hojjatabdollahi** [comment]
@YunYang1994 , the total_loss is now fluctuating between 0.4 to 1.
Now the problem is that we need a piece of code to evaluate the training and calculate the accuracy.

**Comment by tantao258** [comment]
@hojjatabdollahi how long the model have been trained?

**Comment by YunYang1994** [comment]
@tantao258 @hojjatabdollahi I will train it in

**Comment by YunYang1994** [comment]
@hojjatabdollahi `python convert_weight.py -ckpt_file your_trained_yolov3.ckpt --freeze` and `python nms_demo.py` , may you can try it ?  now the question is that I am not very sure of the definition of recall and precision in yolov3.

I have update the training codes, welcome to correct me!

**Comment by YunYang1994** [comment]
Can you compare your loss value with the below loss that is computed by official `yolov3.weights` 
My code is training now. maybe I will show my results in the next week
```
=> EPOCH:   0	| prec_50:1.0000	rec_50:0.6667	avg_iou:0.7794	 | total_loss: 0.1102	loss_coord: 0.0351	loss_sizes: 0.0014	loss_confs: 0.0039	loss_class: 0.0699
=> EPOCH:   1	| prec_50:1.0000	rec_50:1.0000	avg_iou:0.7481	 | total_loss: 0.0703	loss_coord: 0.0081	loss_sizes: 0.0585	loss_confs: 0.0005	loss_class: 0.0032
=> EPOCH:   2	| prec_50:1.0000	rec_50:0.5000	avg_iou:0.8629	 | total_loss: 0.2518	loss_coord: 0.1043	loss_sizes: 0.0182	loss_confs: 0.0087	loss_class: 0.1206
=> EPOCH:   3	| prec_50:1.0000	rec_50:0.5000	avg_iou:0.6165	 | total_loss: 0.0490	loss_coord: 0.0270	loss_sizes: 0.0041	loss_confs: 0.0081	loss_class: 0.0098
=> EPOCH:   4	| prec_50:1.0000	rec_50:0.3333	avg_iou:0.6405	 | total_loss: 1.1871	loss_coord: 0.0141	loss_sizes: 0.0131	loss_confs: 0.0102	loss_class: 1.1498
=> EPOCH:   5	| prec_50:1.0000	rec_50:0.2941	avg_iou:0.7716	 | total_loss: 0.5437	loss_coord: 0.1869	loss_sizes: 0.1315	loss_confs: 0.0223	loss_class: 0.2031
=> EPOCH:   6	| prec_50:1.0000	rec_50:0.2857	avg_iou:0.8808	 | total_loss: 0.4933	loss_coord: 0.0736	loss_sizes: 0.0480	loss_confs: 0.0116	loss_class: 0.3602
=> EPOCH:   7	| prec_50:1.0000	rec_50:0.1667	avg_iou:0.5352	 | total_loss: 0.1554	loss_coord: 0.0382	loss_sizes: 0.0881	loss_confs: 0.0038	loss_class: 0.0253
=> EPOCH:   8	| prec_50:0.0000	rec_50:0.0000	avg_iou:0.7785	 | total_loss: 0.1411	loss_coord: 0.0477	loss_sizes: 0.0191	loss_confs: 0.0013	loss_class: 0.0730
=> EPOCH:   9	| prec_50:1.0000	rec_50:0.7000	avg_iou:0.7605	 | total_loss: 2.7741	loss_coord: 0.1259	loss_sizes: 0.0330	loss_confs: 0.0112	loss_class: 2.6040
=> EPOCH:  10	| prec_50:1.0000	rec_50:0.2500	avg_iou:0.8388	 | total_loss: 1.1068	loss_coord: 0.1270	loss_sizes: 0.1116	loss_confs: 0.0065	loss_class: 0.8616
=> EPOCH:  11	| prec_50:1.0000	rec_50:0.5000	avg_iou:0.7640	 | total_loss: 0.0776	loss_coord: 0.0145	loss_sizes: 0.0423	loss_confs: 0.0121	loss_class: 0.0086
=> EPOCH:  12	| prec_50:1.0000	rec_50:0.5000	avg_iou:0.6321	 | total_loss: 0.1490	loss_coord: 0.0238	loss_sizes: 0.0082	loss_confs: 0.0042	loss_class: 0.1129
=> EPOCH:  13	| prec_50:1.0000	rec_50:0.6667	avg_iou:0.9211	 | total_loss: 4.0471	loss_coord: 0.0338	loss_sizes: 0.0090	loss_confs: 0.0126	loss_class: 3.9917
=> EPOCH:  14	| prec_50:1.0000	rec_50:0.8333	avg_iou:0.8536	 | total_loss: 2.9942	loss_coord: 0.0084	loss_sizes: 0.0297	loss_confs: 0.0022	loss_class: 2.9539
=> EPOCH:  15	| prec_50:1.0000	rec_50:0.4375	avg_iou:0.5469	 | total_loss: 0.8941	loss_coord: 0.1284	loss_sizes: 0.2854	loss_confs: 0.0119	loss_class: 0.4684
=> EPOCH:  16	| prec_50:1.0000	rec_50:0.4286	avg_iou:0.8309	 | total_loss: 0.2362	loss_coord: 0.1163	loss_sizes: 0.0799	loss_confs: 0.0042	loss_class: 0.0358
=> EPOCH:  17	| prec_50:1.0000	rec_50:0.5000	avg_iou:0.5239	 | total_loss: 0.1433	loss_coord: 0.0429	loss_sizes: 0.0436	loss_confs: 0.0061	loss_class: 0.0507
=> EPOCH:  18	| prec_50:1.0000	rec_50:0.5455	avg_iou:0.8923	 | total_loss: 0.4940	loss_coord: 0.0451	loss_sizes: 0.0377	loss_confs: 0.0095	loss_class: 0.4018
=> EPOCH:  19	| prec_50:1.0000	rec_50:0.6667	avg_iou:0.7264	 | total_loss: 1.0627	loss_coord: 0.0723	loss_sizes: 0.0315	loss_confs: 0.0145	loss_class: 0.9444
```

**Comment by muye5** [comment]
> @YunYang1994 , the total_loss is now fluctuating between 0.4 to 1.
> Now the problem is that we need a piece of code to evaluate the training and calculate the accuracy.

Could your share the model evaluating result on coco? thanks.

**Comment by wonder-hy** [comment]
from your code, i think you may confuse "epoch" with "step"(or batch_num)

**Comment by WeifaGan** [comment]
@tantao258 
Hi, I heard that you have trained your model. Are there some errors in the original train.py? And you changed them? Could you give me some advice? Please, Thanks
